tests/blackbox/stratisd_cert.py

```diff
@@ -70,6 +103,206 @@ def test_get_managed_objects(self):
         self.assertEqual(stderr, "")
         self.assertNotEqual(stdout, "")
 
+    def test_stratisd_version(self):
+        """
+        Test getting the daemon version.
+        """
+        self.assertNotEqual(StratisDbus.stratisd_version(), [])
```

Comment:
Here's a good example of where types are important. You documented ```stratisd_version()``` as returning a string, and I think that's quite right. But here you're asserting that it's not equal to ```[]``` which is the empty list. A string can not ever be equal to the empty list, so this test will always succeed, so long as stratisd is alive enough to return a properly formatted GetManagedObjects() result. You want something more precise, but not so precise as to be excessively finicky, i.e., you'll definitely want to avoid string matching using regular expressions. For this test, I'ld suggest something like this:

Try using the method ```assertIsInstance```. There's a related method, ```isinstance``` that behaves similarly.:

```
>>> isinstance
<built-in function isinstance>
>>> isinstance("i am a string", str)
True
>>> isinstance(int(32), int)
True
>>> isinstance(int(32), double)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'double' is not defined
>>> isinstance(int(32), float)
False
>>> isinstance(dbus.String("abc"), str)
True
```
and so forth. Use the method ```assertIsInstance``` to assert that the method is an instance of the correct type. Then, use other assertions to check the value. This way, the test cuts much closer to the boundary between correct and not.

For this particular test, I'ld try:

```
result = StratisDbus.stratisd_version()
self.assertIsInstance(result, str)
self.assertIsNotEqual(result, "")
```